tests/system_tests/test_api_endpoints.py

```diff
@@ -0,0 +1,190 @@
+import time
+from typing import Any
+
+import pytest
+import requests
+from fastapi import status
+from pydantic import TypeAdapter
+
+from blueapi.client.rest import BlueapiRestClient, BlueskyRemoteControlError
+from blueapi.service.interface import get_devices, get_plans
+from blueapi.service.model import (
+    DeviceResponse,
+    EnvironmentResponse,
+    PlanResponse,
+    TasksListResponse,
+    WorkerTask,
+)
+from blueapi.worker.event import TaskStatusEnum, WorkerState
+from blueapi.worker.task import Task
+from blueapi.worker.task_worker import TrackableTask
+
+_SIMPLE_TASK = Task(name="sleep", params={"time": 0.0})
+_LONG_TASK = Task(name="sleep", params={"time": 1.0})
+
+rest = BlueapiRestClient()
+
+
+def get_response(
+    url: str, BaseModel: Any, status_code: int = status.HTTP_200_OK
+) -> Any:
+    get_response = requests.get(url)
+    model = TypeAdapter(BaseModel).validate_python(get_response.json())
+    assert get_response.status_code == status_code
+    return model
+
+
+def test_get_plans():
+    assert rest.get_plans() == PlanResponse(plans=get_plans())
```

Comment:
Should: While this test is useful it is ultimately calling `get_plans()` on both sides of the `==`, so in some cases it boils down to `assert with_extra_steps(get_plans()) == get_plans()`. The test will work fine if there is a problem in the extra steps, but not in every case.

For example, say there is a bug which makes `get_plans()` return `[]`, this test would still pass.

I think the only good solution is to write the list of expected plans yourself, in this module. It will make the test fail if anyone changes the default set but I _think_ that's what we want, then we are quickly made aware of any changes. I welcome opinions/comments, tagging @DiamondJoseph too.

Reply:
I agree that that's what we want to be checking, `with_extra_steps(get_plans()) == <some hardcoded value>`, which we just acknowledge that this test (similarly to the Schema test) may be one that needs to be changed for seemingly unrelated changes, such as changing default loaded modules/changes in those modules.